packages/core/application/core-application-browser/src/application.ts

```diff
@@ -126,17 +101,19 @@ export interface App<HistoryLocationState = unknown> extends AppNavOptions {
   status?: AppStatus;
 
   /**
-   * The initial status of the application's navLink.
-   * Defaulting to `visible` if `status` is `accessible` and `hidden` if status is `inaccessible`
-   * See {@link AppNavLinkStatus}
-   */
-  navLinkStatus?: AppNavLinkStatus;
-
-  /**
-   * The initial flag to determine if the application is searchable in the global search.
-   * Defaulting to `true` if `navLinkStatus` is `visible` or omitted.
+   * Optional list of locations where the app is visible.
+   *
+   * Accepts the following values:
+   * - "globalSearch": the link will appear in the global search bar
+   * - "home": the link will appear on the Kibana home page
+   * - "kibanaOverview": the link will appear in the Kibana overview page
+   * - "sideNav": the link will appear in the side navigation.
+   *   Note: "sideNav" will be deprecated when we change the navigation to "solutions" style.
+   *
+   * @default ['globalSearch', 'sideNav']
+   * unless the status is marked as `inaccessible`.
```

Comment:
nit: maybe we should explicitly mention that we set to `hidden` in that case?

Reply:
Good idea 👍